### DISCUSSION TITLE
All Core Devs Call #14

### DISCUSSION THREAD

**Discussion by ericnakagawa** [celo-org/celo-proposals#395] [CIP Discussions]
# Event Details
When: Friday, August 18, 2023, 11 AM ET/8 AM PST/5 PM CET, 30 minutes

Video call link: https://meet.google.com/shx-pqjw-xzx
Or dial: ‪(US) +1 442-264-7489‬ PIN: ‪575 249 667‬#

# Summary
The Celo All-Core Dev Call is the "All Hands" call for all Celo Core Developers to coordinate technical specifications and proposals and discussions to the Celo Platform.

# Agenda
* See below

Please post all relevant proposals below that should be discussed and evaluated.

**Comment by therealharpaljadeja** [comment]
### Agenda

- Hardfork Timeline update
- [Removal of CIPs 20, 25, 30 & 31](https://github.com/celo-org/celo-proposals/discussions/392)
- Introduction of new transaction type ([Envelope Transaction](https://github.com/celo-org/celo-proposals/discussions/393))
- [Block size limit increase](https://github.com/celo-org/celo-proposals/discussions/394)
- Updated recommended validator specs

**Comment by therealharpaljadeja** [comment]
**Attendees**: Gaston, Samiul, Connor, Javier, Eric, Mariano, Joonkyo, David, Jessan, Subha, Hyunggi, Harpal

**Hardfork timeline update**

August 15: Code complete for all HF features
August 21: Baklava release published
September 1: Baklava upgrade date
September 3: Mainnet release published
September 26: Hard Fork date

- Progress on L2 work
- Additional issues found during testing that can be resolved along in the hardfork update
- Some additional changes that can be clubbed in the hardfork on Baklava
- 3 additional changes are introduced and expanded below.
- We propose to add these changes to another release of the hardfork on Baklava with the above adjusted timeline.

[**Removal of CIPs**](https://github.com/celo-org/celo-proposals/discussions/392)
- The CIPs are precompiles we propose to be removed. 
- These precompiles are not available on Ethereum and are not being used by the Celo ecosystem.
- This will help minimize the difference b/w the Ethereum client and Celo client

[**Introduction of new transaction**](https://github.com/celo-org/celo-proposals/discussions/393)
- The input field related to fee currencies like `gatewayFee` and `gatewayFeeRecipient` will be removed
- We want to introduce a new envelope transaction, and this will be the only envelope transaction available after the L2 upgrade.
- The transaction will be the same as EIP1559 with an additional field `feeCurrency` used for specifying the currency to pay for gas.
- If added and executed along with the Gingerbread Hardfork in September, developers will have the whole Q4 and Q1 of 2024 to migrate
- It also helps us reduce the Ethereum client diff.
 
[**Block Size Limitation**](https://github.com/celo-org/celo-proposals/discussions/394)
- We have been working on increasing throughput and scalability of Celo.
- Few months back, when the block size was 250M, messaging issues b/w nodes stalled the network.
- We propose to increase the block size again in terms of gasLimit, but we have discussed how can we mitigate the problem
- The problem is the message and block sizes which are determined by the usage of calldata we have thought of ideas like changing prices for calldata.
- We believe we should hard code the block size so the block can't practically be bigger than that.
- We propose a hard limit of 5,242,880 (5MB)
- If someone decides to use a lot of calldata the validators will be able to handle it

**Updated Validator Spec**
- Relating to the above point, we propose recommended specs for the validators and nodes.
- Exact specification can be found [here](https://forum.celo.org/t/gingerbread-hard-fork-timeline-specification-update-all-core-devs-call-on-aug-18/6382)
- We have asked validators to update their specs and confirm if they are already on them. [Let us know by responding to this form](https://docs.google.com/forms/d/1WmfRXb9kJ4qHitjYAGgsHwkBH0HOSjBnGY031u8EO_g/viewform?edit_requested=true)

Please feel free to drop questions here and join the discussion.

[Video recording of the call](https://drive.google.com/file/d/1XxBv735vn67wGZmatwHtQPKeMSnvYhCL/view?usp=drive_link) 

Harpalsinh Jadeja
DevRel
Celo Foundation
